src/lib/plugin.ts

```diff
@@ -113,14 +115,35 @@ export const nodeModulesPolyfillPlugin = (options: NodePolyfillsOptions = {}): P
 			const filter = new RegExp(`^(?:node:)?(?:${bundledModules.map(escapeRegex).join('|')})$`);
 
 			const resolver = async (args: OnResolveArgs): Promise<OnResolveResult | undefined> => {
-				const moduleName = normalizeNodeBuiltinPath(args.path);
-
 				const emptyResult = {
 					namespace: emptyNamespace,
 					path: args.path,
 					sideEffects: false,
 				};
 
+				// https://github.com/defunctzombie/package-browser-field-spec
+				if (initialOptions.platform === 'browser') {
+					const packageJson = await loadPackageJSON(args.resolveDir);
+					const browserFieldValue = packageJson?.browser?.[args.path];
+					if (browserFieldValue !== undefined) {
+						// If the module is disabled for browser builds, provide our standard empty module
+						if (browserFieldValue === false) {
+							return emptyResult;
+						}
+
+						// If it's an alias to another module for browser builds, skip resolving the polyfill
+						if (typeof browserFieldValue === 'string') {
+							return;
+						}
+
+						throw new Error(
+							`The "browser" field in package.json (resolved from ${args.resolveDir}) must be a string or false, received ${browserFieldValue}`,
+						);
+					}
```

Comment:
```suggestion
if (browserFieldValue === false || typeof browserFieldValue === 'string') {
						return;
					}
```

let's leave the `false` part to `esbuild` to handle